packages/base-driver/lib/protocol/bidi-commands.js

```diff
@@ -0,0 +1,31 @@
+const SUBSCRIPTION_REQUEST_PARAMS = /** @type {const} */ ({
+  required: ['events'],
+  optional: ['contexts'],
+});
+
+const BIDI_COMMANDS = /** @type {const} */ ({
```

Comment:
You might want to use `@wdio/protocols` which provides a JSON object with various information about all bidi commands. If you require a certain format to be uphold you could reduce the object to what ever shape it needs to be in.

Reply:
oh this is a good idea! i'll have a look

Reply:
I don't see any way to reliably get parameter info from this JSON object, so I'm not sure it will work for us right now. Basically, we'd need to turn this:

```
'<pre>{<br />  events: string[];<br />  contexts?: BrowsingContextBrowsingContext[];<br />}</pre>'
```

into this:

```
{
  required: ['events'],
  optional: ['contexts'],
}
```

And I don't want to resort to regex parsing or TS reflection to do that.

Reply:
@jlipps gotcha, I think this should be possible to do with [`cddl`](https://www.npmjs.com/package/cddl) but requires further work.